web/src/store/index.js

```diff
@@ -226,7 +226,9 @@ const actions = {
 
   async [LOAD_PLOT]({ commit }, { dataset_id, name }) {
     try {
-      const { data } = await CSVService.getPlot(dataset_id, name);
+      const { data } = await CSVService.getPlot(dataset_id, name, {
+        max_components: 6,
```

Comment:
This is a magic number. At minimum probably should pull out as a constant with a relevant name.

Reply:
And more generally, it should probably be user-specifiable.

Reply:
See ab7e4762bae7f5e0df9ec5178d74180392aface5..a3383c3cb98f8334db179e24b6bae12fa3f5701b. These commits abstract out the magic number, and then bind that value to a UI control.